build/README.md

```diff
@@ -52,27 +52,27 @@ with the same name, but an additional extension:
 class CopyBuilder implements Builder {
   final String extension;
 
-  CopyBuilder(this.extension)
+  CopyBuilder(this.extension);
 
   Future build(BuildStep buildStep) async {
     /// Each [buildStep] has a single input.
-    var input = buildStep.inputId;
+    var inputId = buildStep.inputId;
 
     /// Create a new target [AssetId] based on the old one.
     var copy = inputId.addExtension(extension);
-    var contents = await buildStep.readAsString(input);
+    var contents = await buildStep.readAsString(inputId);
 
     /// Write out the new asset.
     ///
     /// There is no need to `await` here, the system handles waiting on these
     /// files as necessary before advancing to the next phase.
-    buildStep.writeAsString(copy, contents);
+    await buildStep.writeAsString(copy, contents);
```

Comment:
See 2 lines above - "There is no need to `await` here"

The reason this returns a `Future` is so it can implement `AssetReader`, but in this usage it isn't necessary to wait for the write to finish since it doesn't have an effect that impacts you.

If we had a way we'd tell the linter to ignore this.

Reply:
Fixed here

See https://github.com/dart-lang/build/issues/1254